src/DataViews/FunctionsDataView.cpp

```diff
@@ -244,78 +242,55 @@ void FunctionsDataView::OnContextMenu(const std::string& action, int menu_index,
 }
 
 void FunctionsDataView::DoFilter() {
-  m_FilterTokens = absl::StrSplit(absl::AsciiStrToLower(filter_), ' ');
+  filter_tokens_ = absl::StrSplit(absl::AsciiStrToLower(filter_), ' ');
 
-#ifdef WIN32
-  ParallelFilter();
-#else
-  // TODO: port parallel filtering
-  std::vector<uint64_t> indices;
-  const std::vector<const FunctionInfo*>& functions(functions_);
-  for (size_t i = 0; i < functions.size(); ++i) {
-    const FunctionInfo* function = functions[i];
-    std::string name =
-        absl::AsciiStrToLower(orbit_client_data::function_utils::GetDisplayName(*function)) +
-        orbit_client_data::function_utils::GetLoadedModuleName(*function);
-
-    bool match = true;
-
-    for (std::string& filter_token : m_FilterTokens) {
-      if (name.find(filter_token) == std::string::npos) {
-        match = false;
-        break;
-      }
-    }
+  const size_t number_of_threads_available = thread_pool_->GetPoolSize();
 
-    if (match) {
-      indices.push_back(i);
-    }
-  }
+  constexpr size_t kMinimumNumberOfFunctionsPerThread = 64;
```

Comment:
should this be a function of number of cores? I wonder how it works for say 100 core machine.

Reply:
I don't think that this number should depend on the number of cores, rather on the amount of overhead involved in each task.